tests/core/test_file_utils.py

```diff
@@ -9,12 +9,12 @@
 @pytest.mark.parametrize(
     ["filename", "input_content", "expected_content"],
     [
-        ("normal.txt", "Normal", "Normal"),
-        ("0-inside.txt", "Ins\0de", "Ins de"),
+        ("normal.txt", b"Normal", "Normal"),
+        ("invalid-utf8-start-byte.txt", b"Hello\x81World", "Hello\uFFFDWorld"),
```

Comment:
Nit : We could add a test to verify that `\0` is indeed not replaced

Reply:
Good point, just added a case for it